src/main/java/com/vestmark/nexus/plugin/maven/MavenResource.java

```diff
@@ -107,6 +107,9 @@ public Response download(
     try {
       storageTx = storageTxSupplier.get();
       storageTx.begin();
+      if (!storageTx.getDb().isActiveOnCurrentThread()) {
+        storageTx.getDb().activateOnCurrentThread();
+      }
```

Comment:
Do we need to ever deactivate this?

Reply:
I'm a little worried that we're going to start stomping on each other WRT resource contention...but this is worth a shot.